engine/table/src/main/java/io/deephaven/engine/table/impl/perf/QueryPerformanceRecorder.java

```diff
@@ -714,4 +429,63 @@ public static boolean setCallsite() {
     public static void clearCallsite() {
         cachedCallsite.remove();
     }
+
+    /**
+     * Finish the nugget and clear the callsite if needed.
+     *
+     * @param nugget an optional nugget
+     * @param needClear true if the callsite needs to be cleared
+     */
+    private static void finishAndClear(@Nullable final QueryPerformanceNugget nugget, final boolean needClear) {
+        if (nugget != null) {
+            nugget.done();
+        }
+
+        if (needClear) {
+            clearCallsite();
+        }
+    }
+
+    /**
+     * Dummy recorder for use when no recorder is installed.
+     */
+    private static class DummyQueryPerformanceRecorder extends QueryPerformanceRecorder {
```

Comment:
We might be able to streamline this into an impl that always throws UOE for most ops.